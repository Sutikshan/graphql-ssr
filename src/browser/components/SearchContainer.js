import React, { Component } from "react";
import SearchModel from "./SearchModel";
import SearchMake from "./SearchMake";
import { Link } from "react-router-dom";

class SearchContainer extends Component {
  constructor(props) {
    super(props);
    this.state = { makeId: 0 };
    this.onMakeSelect = this.onMakeSelect.bind(this);
    this.onModelSelect = this.onModelSelect.bind(this);
  }

  onMakeSelect(event) {
    this.setState({ makeId: Number.parseInt(event.target.value, 10) });
  }

  onModelSelect(event) {
    this.setState({ modelId: Number.parseInt(event.target.value, 10) });
  }

  render() {
    return (
      <div>
        <h1>Browse Catalog</h1>
        <div className="search-container">
          <SearchMake
            onMakeSelect={this.onMakeSelect}
            makeId={this.state.makeId}
          />
          <SearchModel
            onModelSelect={this.onModelSelect}
            makeId={this.state.makeId}
            modelId={this.state.modelId}
          />
          <div className="button-group">
            {this.state.modelId ? (
              <Link to={`/make/model/${this.state.modelId}`}>
                <button className="search-button">Details</button>
              </Link>
            ) : (
              <button disabled className="search-button disabled">
                Details
              </button>
            )}
            <Link to={`/`}>
              <button className="search-button">Home</button>
            </Link>
          </div>
        </div>
      </div>
    );
  }
}

export default SearchContainer;
