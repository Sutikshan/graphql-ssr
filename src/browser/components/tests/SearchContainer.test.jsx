import React from "react";
import { shallow } from "enzyme";
import SearchContainer from "../SearchContainer";

it("should render a SearchContainer component correctly", () => {
  const wrapper = shallow(<SearchContainer />);

  expect(wrapper).toMatchSnapshot();
});
