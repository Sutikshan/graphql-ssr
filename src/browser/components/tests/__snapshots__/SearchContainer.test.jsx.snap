// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a SearchContainer component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Browse Catalog
        </h1>,
        <div
          className="search-container"
        >
          <Apollo(SearchMake)
            makeId={0}
            onMakeSelect={[Function]}
          />
          <Apollo(SearchModel)
            makeId={0}
            modelId={undefined}
            onModelSelect={[Function]}
          />
          <div
            className="button-group"
          >
            <button
              className="search-button disabled"
              disabled={true}
            >
              Details
            </button>
            <Link
              replace={false}
              to="/"
            >
              <button
                className="search-button"
              >
                Home
              </button>
            </Link>
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Browse Catalog",
        },
        "ref": null,
        "rendered": "Browse Catalog",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Apollo(SearchMake)
              makeId={0}
              onMakeSelect={[Function]}
            />,
            <Apollo(SearchModel)
              makeId={0}
              modelId={undefined}
              onModelSelect={[Function]}
            />,
            <div
              className="button-group"
            >
              <button
                className="search-button disabled"
                disabled={true}
              >
                Details
              </button>
              <Link
                replace={false}
                to="/"
              >
                <button
                  className="search-button"
                >
                  Home
                </button>
              </Link>
            </div>,
          ],
          "className": "search-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "makeId": 0,
              "onMakeSelect": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "makeId": 0,
              "modelId": undefined,
              "onModelSelect": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="search-button disabled"
                  disabled={true}
                >
                  Details
                </button>,
                <Link
                  replace={false}
                  to="/"
                >
                  <button
                    className="search-button"
                  >
                    Home
                  </button>
                </Link>,
              ],
              "className": "button-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Details",
                  "className": "search-button disabled",
                  "disabled": true,
                },
                "ref": null,
                "rendered": "Details",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <button
                    className="search-button"
                  >
                    Home
                  </button>,
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Home",
                    "className": "search-button",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": "button",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Browse Catalog
          </h1>,
          <div
            className="search-container"
          >
            <Apollo(SearchMake)
              makeId={0}
              onMakeSelect={[Function]}
            />
            <Apollo(SearchModel)
              makeId={0}
              modelId={undefined}
              onModelSelect={[Function]}
            />
            <div
              className="button-group"
            >
              <button
                className="search-button disabled"
                disabled={true}
              >
                Details
              </button>
              <Link
                replace={false}
                to="/"
              >
                <button
                  className="search-button"
                >
                  Home
                </button>
              </Link>
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Browse Catalog",
          },
          "ref": null,
          "rendered": "Browse Catalog",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Apollo(SearchMake)
                makeId={0}
                onMakeSelect={[Function]}
              />,
              <Apollo(SearchModel)
                makeId={0}
                modelId={undefined}
                onModelSelect={[Function]}
              />,
              <div
                className="button-group"
              >
                <button
                  className="search-button disabled"
                  disabled={true}
                >
                  Details
                </button>
                <Link
                  replace={false}
                  to="/"
                >
                  <button
                    className="search-button"
                  >
                    Home
                  </button>
                </Link>
              </div>,
            ],
            "className": "search-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "makeId": 0,
                "onMakeSelect": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "makeId": 0,
                "modelId": undefined,
                "onModelSelect": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="search-button disabled"
                    disabled={true}
                  >
                    Details
                  </button>,
                  <Link
                    replace={false}
                    to="/"
                  >
                    <button
                      className="search-button"
                    >
                      Home
                    </button>
                  </Link>,
                ],
                "className": "button-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Details",
                    "className": "search-button disabled",
                    "disabled": true,
                  },
                  "ref": null,
                  "rendered": "Details",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <button
                      className="search-button"
                    >
                      Home
                    </button>,
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Home",
                      "className": "search-button",
                    },
                    "ref": null,
                    "rendered": "Home",
                    "type": "button",
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
